plugin/src/main/java/org/gradle/testretry/internal/framework/TestNgTestFrameworkStrategy.java

```diff
@@ -59,7 +61,7 @@ public TestFramework createRetrying(JvmTestExecutionSpec spec, Test testTask, Se
                 }
             });
 
-        return new TestNGTestFramework(testTask, retriedTestFilter, instantiator, classLoaderCache);
+        return new TestNGTestFramework(testTask, ((ProjectInternal)testTask.getProject()).getServices().get(ObjectFactory.class).fileCollection(), retriedTestFilter, ((ProjectInternal)testTask.getProject()).getServices().get(ObjectFactory.class));
```

Comment:
❓ Why was this change necessary? `ObjectFactory` could be extracted as a local variable to increase readability.

Reply:
This change was necessary to be compatible with Gradle 6.6. But it actually breaks backwards compatibility with previous Gradle versions. I think we should aim for backwards compatibility here and extend the implementation in that regard.

Reply:
I agree. We should add `@UsedByScanPlugin("test-retry")` annotations to all internal APIs that we rely on so this doesn't go unnoticed next time.